client/components/themes-list/index.jsx

```diff
@@ -109,11 +108,123 @@ export const ThemesList = React.createClass( {
 
 		return (
 			<div className="themes-list">
-				{ this.props.themes.map( this.renderTheme ) }
-				{ this.props.loading && this.renderLoadingPlaceholders() }
-				{ this.renderTrailingItems() }
+				<InfiniteLoader
+					isRowLoaded={ this.isRowLoaded }
+					loadMoreRows={ this.loadMoreRows }
+					rowCount={ 1000 }
+					threshold={ 20 }
+				>
+					{ ( { onRowsRendered, registerChild } ) => (
+						<WindowScroller ref={ this.onScrollerRendered } onScroll={ this.onScroll }>
+							{ ( { scrollTop } ) => {
+								return this.renderGrid( scrollTop, onRowsRendered, registerChild );
+							} }
+						</WindowScroller>
+					) }
+				</InfiniteLoader>
 			</div>
 		);
+	},
+
+	renderGrid( scrollTop, onRowsRendered, registerChild ) {
+		const minColumnWidth = 250;
+		const ssrSpacers = 20;
+
+		const columnCount = this._width === undefined ? 1 : Math.floor( this._width / minColumnWidth );
+		const rowCount = this._width === undefined
+			? this.props.themes.length + ssrSpacers
+			: Math.ceil( this.props.themesCount / columnCount );
+
+		const columnWidth = Math.floor( this._width / columnCount );
+		const rowHeight = Math.floor( ( columnWidth - 20 ) * 0.75 ) + 74;
+
+		const onCellsRendered = ( { stopIndex } ) => {
+			if (
+				! this.props.loading &&
+				! this.isRowLoaded( stopIndex )
+			) {
+				onRowsRendered( {
+					startIndex: Math.min( this.props.themes.length, stopIndex ),
+					stopIndex: Math.min( this.props.themes.length + 19, this.props.themesCount - 1, stopIndex )
```

Comment:
I added a self-explanatory constant for `19` and a more elaborate explanation for what's happening in the `rowHeight` calculation.